tests/spark/autologging/ml/test_pyspark_ml_autologging.py

```diff
@@ -876,3 +898,237 @@ def test_autolog_registering_model(spark_session, dataset_binomial):
+        ]
+    ).toDF("id", "category", "not_a_feature_1", "not_a_feature_2")
+
+
+@pytest.fixture()
+def pipeline_for_feature_cols(input_df_with_non_features):
+    string_indexer = StringIndexer(inputCol="category", outputCol="label").fit(
+        input_df_with_non_features
+    )
+    return Pipeline(
+        stages=[
+            string_indexer,
+            VectorAssembler(inputCols=["id"], outputCol="features"),
+            LogisticRegression(),
+            IndexToString(
+                inputCol="prediction", outputCol="originalLabel", labels=string_indexer.labels
+            ),
+        ]
+    )
+
+
+@pytest.mark.large
+def test_signature_with_non_feature_input_columns(
+    input_df_with_non_features, pipeline_for_feature_cols
+):
+    mlflow.pyspark.ml.autolog(log_models=True, log_input_examples=True)
+    with mlflow.start_run() as run:
+        pipeline_for_feature_cols.fit(input_df_with_non_features)
+        _assert_autolog_infers_model_signature_correctly(
+            run,
+            [{"name": "id", "type": "long"}],
+            [
+                {"name": "features", "type": "string"},
+                {"name": "rawPrediction", "type": "string"},
+                {"name": "probability", "type": "string"},
+                {"name": "prediction", "type": "double"},
+                {"name": "originalLabel", "type": "string"},
+            ],
+        )
+
+
+@pytest.mark.large
+def test_spark_df_with_vector_to_array_casts_successfully(dataset_multinomial):
+    from pyspark.sql.types import ArrayType, DoubleType
+
+    output_df = cast_spark_df_with_vector_to_array(dataset_multinomial)
+    assert [_field for _field in output_df.schema.fields if _field.name == "features"][
+        0
+    ].dataType == ArrayType(
+        DoubleType(), False
+    ), "'features' column isn't of expected type array<double>"
+
+
+@pytest.mark.large
+def test_get_feature_cols(input_df_with_non_features, pipeline_for_feature_cols):
+    pipeline_model = pipeline_for_feature_cols.fit(input_df_with_non_features)
+    assert get_feature_cols(input_df_with_non_features.schema, pipeline_model) == {
+        "id"
+    }, "Wrong feature columns returned"
+
+
+@pytest.mark.large
+def test_find_and_set_features_col_as_vector_if_needed(lr, dataset_binomial):
+    from mlflow.spark import find_and_set_features_col_as_vector_if_needed
+    from pyspark.ml.linalg import VectorUDT
+    from pyspark.sql.utils import IllegalArgumentException
+
+    pipeline_model = lr.fit(dataset_binomial)
+    df_with_array_features = cast_spark_df_with_vector_to_array(dataset_binomial)
+    df_with_vector_features = find_and_set_features_col_as_vector_if_needed(
+        df_with_array_features, pipeline_model
+    )
+    assert isinstance(
+        [_field for _field in df_with_vector_features.schema.fields if _field.name == "features"][
+            0
+        ].dataType,
+        VectorUDT,
+    ), "'features' column wasn't cast to vector type"
```

Comment:
Can we split this into two lines?

```suggestion
    features_col = next(
        filter(lambda f: f.name == "features", df_with_vector_features.schema.fields)
    )
    assert isinstance(features_col.dataType, VectorUDT)

```